core/container/src/main/java/org/wildfly/swarm/container/runtime/usage/NetworkVariableSupplier.java

```diff
@@ -0,0 +1,92 @@
+package org.wildfly.swarm.container.runtime.usage;
+
+import org.wildfly.swarm.container.Interface;
+import org.wildfly.swarm.spi.api.SocketBinding;
+import org.wildfly.swarm.spi.api.SocketBindingGroup;
+
+/**
+ * Created by bob on 9/7/17.
+ */
+public class NetworkVariableSupplier implements UsageVariableSupplier {
+
+    public NetworkVariableSupplier(Iterable<Interface> interfaces, Iterable<SocketBindingGroup> socketBindings, UsageVariableSupplier delegate) {
+        this.interfaces = interfaces;
+        this.socketBindings = socketBindings;
+        this.delegate = delegate;
+    }
+
+    @Override
+    public Object valueOf(String name) throws Exception {
+        // very special case
+        if (name.equals("swarm.public.url.base")) {
+            return "http://" + valueOf("swarm.public.host") + ":" + valueOf("swarm.http.port") + "/";
+
+        }
+        String[] parts = name.split("\\.");
+        if (parts.length == 3) {
+            for (Interface each : this.interfaces) {
+                if (parts[1].equals(each.getName())) {
+                    if (parts[2].equals("host")) {
+                        return each.getExpression();
+                    }
+                }
+            }
+        }
```

Comment:
Is it guaranteed that `parts[0]` is always `"swarm"`?

Reply:
Ah, we should probably confirm that it is.